ccpp/data/GFS_typedefs.F90

```diff
@@ -5800,6 +5804,10 @@ subroutine diag_create (Diag, IM, Model)
     !--- 3D diagnostics for Thompson MP / GFDL MP
     allocate (Diag%refl_10cm(IM,Model%levs))
 
+    !--- New PBL Diagnostics
+    allocate (Diag%dkt(IM,Model%levs))
```

Comment:
Shouldn't these be allocated to `Model%levs-1` to match the previous allocation in the interstitial DDT? If not, i.e. if you want to use `Model%levs` (for example to avoid issues when writing out the diagnostics), then you need to change the `dimensions` metadata in fv3atm and ccpp-physics, and the array declarations (the lines with `intent(in)`) from `dku(im,km-1)` to `dku(im,km)` (same for `dkt`) in both moninemf and satmedmfvdifq (and satmedmfvdif or any other scheme that is using either of these variables) .

Reply:
Thanks, @climbfuji! Good catch. Indeed, it should be Model%levs-1. Changing it right now. 

In the meantime, I do see some of the atmosphere_heat/momentum_diffusivity* variables use Model%levs, while others use Model%levs-1. Probably also something keep in mind for future code standardization/unification.

Thanks!

Reply:
Yes, please create an issue so that it doesn't get forgotten (unless you can remember, I certainly cannot). Does the diagnostic output still work without crashing, also in debug mode, when making this change?

Reply:
@climbfuji, Sure, I will create an issue either in FV3atm or ccpp/physics (please let me where you would prefer to open this issue, Dom) for potentially unifying the usage of different flavors (names, dimensions, etc.) of atmosphere_heat/momentum_diffusivity variables in FV3atm and ccpp/physics levels. Maybe this should apply to other variables as well. But, this will need some infrastructure level design/standard updates/changes.

For the latest feature/dkudiagnostic branch, @ChunxiZhang-NOAA is doing model-level regression tests now on Hera. Let's see how it goes with the latest changes. And Chunxi, please make sure you update your local branch with these latest changes. Thanks!

Reply:
@BinLiu-NOAA thinking about this more, since there is no regression test for this, you will need to test manually: compile in debug mode, use an existing regression test that uses satmedmfvdifq or moninedmf, add the two arrays dku and dkt to `diag_table`, check if the code runs w/o out-of-bounds errors. Not clear how this will look like in the diagnostic files if these two arrays have one level less in the vertical dimension.

Reply:
@climbfuji, I have just brought this branch up to date with the latest develop branch. Let's first test to make sure the existing model-level RTs can go through. For the manual test you mentioned, could @ChunxiZhang-NOAA or @AndrewHazelton help to conduct that test. Not sure if we will be able to finish the above manual test you mentioned today though. Thanks!

Reply:
I am planning to test the update to fix the convective scheme issue in the global-nested version soon, so I can test dku and dkt then as well if you want.

Reply:
Thanks @ChunxiZhang-NOAA! For the full model-level regression tests, I believe @BrianCurtis-NOAA is currently conducting that on Hera, I am doing it on Orion. So, you probably only need to conduct this specific test you and @climbfuji just mentioned to output the dku/dkt, maybe also test it under the debug mode. Thanks!

Reply:
@BinLiu-NOAA  Sure. I just started working on this specific test on Hera.

Reply:
I added dkt and dku in the diag table, the outputs looks fine except there are mixed zero and missing values at the top model level 64.  The run for the debug mode failed. The error message is: forrtl: severe (408): fort: (2): Subscript #2 of the array VAR3 has value 64 which is greater than the upper bound of 63.   We can ask Jongil if he would like to change the vertical dimension of dkt and dku from km-1 to km. I think he moved one level down to make all km-1 levels have values because the first level is zero.

Reply:
That's what I expected to happen, thanks for confirming this @ChunxiZhang-NOAA. I think it is a good idea to extend `dku` and `dkt` to `km` in the vertical. It gets initialized to zero and stays zero for that level all the time.

Reply:
@ChunxiZhang-NOAA Thanks for testing this out! And @climbfuji, it looks to me that we should define and allocate dkt/dku at the FV3atm level as km, while leave them as km-1 in the ccpp/physics level. In this case, at the FV3atm level it will be properly defined, allocated, and initialized for all km levels. In the ccpp/physics level, only those km-1 levels will be used/updated. I suspect this probably will work fine, even with the debug mode. What do you think?

Reply:
Hi @climbfuji, @junwang-noaa, @BrianCurtis-NOAA, Following the above discussions, I made the following changes at the FV3atm level:

- Define and allocate dkt and dku (atmosphere heat/momentum diffusivity in GFS_typedefs.F90/meta with km levels.
- Unify the long name and Diag%desc for dkt and dku (address Jun's another comment)

See more details can be seen from [this link](https://github.com/hafs-community/fv3atm/compare/feature/dkudiagnostic...hafs-community:feature/dktdku?expand=1). 
@ChunxiZhang-NOAA conducted a test with his previous failed debug test when outputting dkt and dku as diag variables, which can now go through successfully.

If you all agree, I can commit these changes into this Pull Request. Please let us know how you would like to proceed. Thanks!